### PR TITLE
Add Accordion Types

### LINKED ISSUE #3601: Add Accordion TypeScript Types
As mentioned in #3409, the Accordion component is missing TS types. In order for this issue to be resolved, we'll need a TS type for the following components:

- [ ] Accordion
- [ ] AccordionToggle
- [ ] AccordionCollapse

### PR DESCRIPTION
**Pull request by bpas247**
Fixes #3601

### CODE REVIEW

File: types/simple.test.tsx [hunk 1]
```diff
@@ -2,6 +2,7 @@ import * as React from 'react';
 
 import {
   Alert,
+  Accordion,
```
Comment (bpas247) [inline types/simple.test.tsx]: This line is throwing the following error
```
ERROR: 5:3   expect     TypeScript@next compile error: 
Module '"."' has no exported member 'Accordion'.
```
I feel like I'm missing something obvious here, I'm not sure what it is though...

Comment (mxschmitt) [reply types/simple.test.tsx]: Try to add it also there: https://github.com/react-bootstrap/react-bootstrap/blob/fbc26bbb817629aeb8a6be953603e0e26e88189b/types/components/index.d.ts

Then it should be available for importing.

File: types/components/AccordionToggle.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import Button from './Button';
+import { BsPrefixComponent } from './helpers';
+
+export interface AccordionToggleProps {
+  eventKey: string;
+  onClick?: (arg0?: React.SyntheticEvent) => any;
```
Comment (taion) [inline types/components/AccordionToggle.d.ts]:
```suggestion
  onClick?: (event?: React.SyntheticEvent) => void;
```

File: types/components/AccordionToggle.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+declare class AccordionToggle<
+  As extends React.ReactType = typeof Button
```
Comment (taion) [inline types/components/AccordionToggle.d.ts]: currently this is actually `'button'`, not our `Button`

Comment (bpas247) [reply types/components/AccordionToggle.d.ts]: in the accordion example we're using `Button`
https://github.com/react-bootstrap/react-bootstrap/blob/8819727d423684bb69b0b45dd7b4a97a77d715f3/www/src/examples/Accordion/Basic.js#L4

Comment (taion) [reply types/components/AccordionToggle.d.ts]: right, but the actual default is `button`: https://github.com/react-bootstrap/react-bootstrap/blob/8819727d423684bb69b0b45dd7b4a97a77d715f3/src/AccordionToggle.js#L23

though as noted in that PR, we should consider changing this to `<SafeAnchor>`

**Pull request merged**